pyproject.toml

```diff
@@ -21,10 +21,10 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.7, >= 3.7.2"
-opentelemetry-api = "1.22.0"
-opentelemetry-sdk = "1.22.0"
-opentelemetry-exporter-otlp = "1.22.0"
-opentelemetry-instrumentation = "0.43b0"
+opentelemetry-api = "^1.22.0"
+opentelemetry-sdk = "^1.22.0"
+opentelemetry-exporter-otlp = "^1.22.0"
+opentelemetry-instrumentation = "^0.43b0"
```

Comment:
Should we keep the instrumentation at a tilde instead of a caret to help minimize dependency conflicts since it's a 0.x instead of a 1.x?

Reply:
It looks like `^` has special handling for 0.x releases only allow [patch updates](https://python-poetry.org/docs/dependency-specification/#caret-requirements) so I think this is fine.

Reply:
Ah, you're right. We don't have a patch version in our dep so need the `~` too. Updated 👍🏻